packages/app/src/api.ts

```diff
@@ -235,32 +235,37 @@ export function initializeServerApp(
     throw ERROR_FACTORY.create(AppError.INVALID_SERVER_APP_ENVIRONMENT);
   }
 
+  if (_serverAppConfig.automaticDataCollectionEnabled === undefined) {
```

Comment:
Previous detection of the resuable FirebaseServerApps weren't taking the automaticDataCollectionEnabled configuration option into account. We need to explictly set this to false if the optional value is not defined.